### PR TITLE
ACCSEC-22010 challenges order

### PR DESCRIPTION
**Pull request by yafuquen**
## Jira Ticket
- ACCSEC-22010

## Description
- Add order parameter to get challenge list
- Add silently approve challenges per factor

## Commit message
- feat: Ordering for challenge list

## Screenshot
Screenshot_1630351811

## Testing
- [ ] Added unit tests
- [x] Ran unit tests successfully
- [x] Added documentation for public APIs and/or Wiki

### CODE REVIEW

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Fix code convention problems</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request title does not follow the convention [ACCSEC-0000] PR Title text</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request title does not follow the convention [ACCSEC-0000] PR Title text</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request title does not follow the convention [ACCSEC-0000] PR Title text</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request title does not follow the convention [ACCSEC-0000] PR Title text</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: sample/src/main/java/com/twilio/verify/sample/viewmodel/ChallengesViewModel.kt [hunk 1]
```diff
@@ -30,10 +31,9 @@ class ChallengesViewModel(private val twilioVerifyAdapter: TwilioVerifyAdapter)
 
   fun loadChallenges(factorSid: String) {
     twilioVerifyAdapter.getAllChallenges(
-      ChallengeListPayload(factorSid, PAGE_SIZE),
+      ChallengeListPayload(factorSid, PAGE_SIZE, order = Desc),
```
Comment (AlejandroOrozco) [inline sample/src/main/java/com/twilio/verify/sample/viewmodel/ChallengesViewModel.kt:34]: Should the default it be Ascending or Descending ?

https://github.com/twilio/twilio-verify-ios/blob/b19e2824168a877b00a9c720bc189580fb23e600/TwilioVerifySDK/TwilioVerify/Sources/Models/ChallengeListPayload.swift#L45

Comment (yafuquen) [reply sample/src/main/java/com/twilio/verify/sample/viewmodel/ChallengesViewModel.kt:34]: To keep the previous ordering for customers already integrated and as backend default value is asc, we should use asc

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request title does not follow the convention [ACCSEC-0000] PR Title text</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by SergioFierro — APPROVED** [review]

**Review by AlejandroOrozco — APPROVED** [review]

**Pull request merged**